com.unity.render-pipelines.core/Runtime/Textures/RTHandleSystem.cs

```diff
@@ -221,19 +226,54 @@ public void SetReferenceSize(int width, int height, MSAASamples msaaSamples, boo
-                var scaledFinalViewport = DynamicResolutionHandler.instance.ApplyScalesOnSize(DynamicResolutionHandler.instance.finalViewport);
-                var scaledMaxSize = DynamicResolutionHandler.instance.ApplyScalesOnSize(maxSize);
-                float xScale = (float)scaledFinalViewport.x / (float)scaledMaxSize.x;
-                float yScale = (float)scaledFinalViewport.y / (float)scaledMaxSize.y;
-                m_RTHandleProperties.rtHandleScale = new Vector4(xScale, yScale, m_RTHandleProperties.rtHandleScale.x, m_RTHandleProperties.rtHandleScale.y);
+                var scales = CalculateScaledRatioAgainstMaxSize(DynamicResolutionHandler.instance.finalViewport);
+                m_RTHandleProperties.rtHandleScale = new Vector4(scales.x, scales.y, m_RTHandleProperties.rtHandleScale.x, m_RTHandleProperties.rtHandleScale.y);
             }
             else
             {
-                Vector2 maxSize = new Vector2(GetMaxWidth(), GetMaxHeight());
-                Vector2 scaleCurrent = m_RTHandleProperties.currentViewportSize / maxSize;
+                var scales = CalculateRatioAgainstMaxSize(m_RTHandleProperties.currentViewportSize);
                 Vector2 scalePrevious = m_RTHandleProperties.previousViewportSize / lastFrameMaxSize;
-                m_RTHandleProperties.rtHandleScale = new Vector4(scaleCurrent.x, scaleCurrent.y, scalePrevious.x, scalePrevious.y);
+                m_RTHandleProperties.rtHandleScale = new Vector4(scales.x, scales.y, scalePrevious.x, scalePrevious.y);
+            }
+
+            SetPostProcessScale(m_RTHandleProperties.currentViewportSize.x, m_RTHandleProperties.currentViewportSize.y);
+        }
+
+        internal void SetPostProcessScale(int width, int height)
+        {
+            Vector2 scales;
+            if (DynamicResolutionHandler.instance.HardwareDynamicResIsEnabled() && m_HardwareDynamicResRequested)
+            {
+                scales = CalculateScaledRatioAgainstMaxSize(DynamicResolutionHandler.instance.finalViewport);
+            }
+            else
+            {
+                scales = CalculateRatioAgainstMaxSize(width, height);
             }
+            m_RTHandleProperties.rtHandlePostProcessScale = new Vector4(scales.x, scales.y, 0.0f, 0.0f);
+        }
+
+        internal Vector2 CalculateScaledRatioAgainstMaxSize(in Vector2Int viewport)
+        {
+            return CalculateScaledRatioAgainstMaxSize(viewport.x, viewport.y);
+        }
+
+        internal Vector2 CalculateRatioAgainstMaxSize(in Vector2Int viewport)
+        {
+            return CalculateRatioAgainstMaxSize(viewport.x, viewport.y);
+        }
+
+        internal Vector2 CalculateRatioAgainstMaxSize(int width, int height)
+        {
+            Vector2 maxSize = new Vector2Int(GetMaxWidth(), GetMaxHeight());
+            return new Vector2((float)width / maxSize.x, (float)height / maxSize.y);
+        }
+
+        internal Vector2 CalculateScaledRatioAgainstMaxSize(int width, int height)
```

Comment:
I don't think we need one overload Vector2Int and one with the two separate, I'd keep only the Vector2Int , but either are fine I think

Reply:
Got it